lib/screens/qr_code_screen.dart

```diff
@@ -0,0 +1,83 @@
+import 'dart:async';
+import 'package:flutter/material.dart';
+import 'package:barcode_widget/barcode_widget.dart';
+
+class QrCodeScreen extends StatefulWidget {
+  final String data;
+  final String deviceId;
+  final int time;
+  const QrCodeScreen({Key? key, required this.data, required this.deviceId, required this.time}) : super(key: key);
+
+  @override
+  _QrCodeState createState() => _QrCodeState();
+}
+
+class _QrCodeState extends State<QrCodeScreen> {
```

Comment:
**question:** do you see any room for improvements on this screen related to redrawing the UI?

Reply:
Yes, I should have used Provider and have changeNotifier to only update the QrCode on new incoming data rather then rebuilding the entire screen